vscode-dotnet-runtime-library/src/Acquisition/DotnetCoreAcquisitionWorker.ts

```diff
@@ -88,39 +112,60 @@ export class DotnetCoreAcquisitionWorker implements IDotnetCoreAcquisitionWorker
         return undefined;
     }
 
-    private async acquire(version: string, installRuntime: boolean): Promise<IDotnetAcquireResult> {
+    /**
+     *
+     * @param version the version to get of the runtime or sdk.
+     * @param installRuntime true for runtime acquisition, false for SDK.
+     * @param global false for local install, true for global SDK installs.
+     * @returns the dotnet acqusition result.
+     */
+    private async acquire(version: string, installRuntime: boolean, globalInstallerResolver : GlobalInstallerResolver | null = null): Promise<IDotnetAcquireResult> {
         const existingAcquisitionPromise = this.acquisitionPromises[version];
-        if (existingAcquisitionPromise) {
+        if (existingAcquisitionPromise)
+        {
             // This version of dotnet is already being acquired. Memoize the promise.
             this.context.eventStream.post(new DotnetAcquisitionInProgress(version));
             return existingAcquisitionPromise.then((res) => ({ dotnetPath: res }));
-        } else {
+        }
+        else
+        {
             // We're the only one acquiring this version of dotnet, start the acquisition process.
-            const acquisitionPromise = this.acquireCore(version, installRuntime).catch((error: Error) => {
-                delete this.acquisitionPromises[version];
-                throw new Error(`.NET Acquisition Failed: ${error.message}`);
-            });
+            let acquisitionPromise = null;
+            if(globalInstallerResolver !== null)
+            {
+                // We are requesting a global sdk install.
+                acquisitionPromise = this.acquireGlobalCore(globalInstallerResolver).catch((error: Error) => {
+                    delete this.acquisitionPromises[version];
+                    throw new Error(`.NET Acquisition Failed: ${error.message}`);
```

Comment:
I was originally going to say that we could deduplicate the error-handling branches of this promise and the local-version promise, but then I had a more scary thought:

the acquisition promises are kept in a map keyed by _version_ only - so it's possible that a request for a global install of version X and a local install of version X would clash. I think we need to key the promises cache by install _type_ as well to prevent this potential clash.

Reply:
Youre right, that is a really good observation. Thank you so much for catching this! I prefixed the global ones with global- because I dont want to break detection of existing local install promises.